### PR TITLE
Fix #1492: Add PIPX_GLOBAL_(HOME|BIN_DIR|MAN_DIR) documentation and list them in `pipx environment`

### LINKED ISSUE #1492: PIPX_GLOBAL_(HOME|BIN_DIR|MAN_DIR) missing from documentation and `pipx environment`
**Describe the bug**

`pipx environment` does not list `PIPX_GLOBAL_(HOME|BIN_DIR|MAN_DIR)` under `Environment variables (set by user)`. Additionally these variables are missing from the documentation of the `environment` and `install` commands, which makes them pretty hard to discover as a user.

**How to reproduce**

```sh
$ pipx environment
Environment variables (set by user):

PIPX_HOME=
PIPX_BIN_DIR=
PIPX_MAN_DIR=
PIPX_SHARED_LIBS=
PIPX_DEFAULT_PYTHON=
PIPX_FETCH_MISSING_PYTHON=
USE_EMOJI=

Derived values (computed by pipx):

PIPX_HOME=/home/user/.local/share/pipx
PIPX_BIN_DIR=/home/user/.local/bin
PIPX_MAN_DIR=/home/user/.local/share/man
PIPX_SHARED_LIBS=/home/user/.local/share/pipx/shared
PIPX_LOCAL_VENVS=/home/user/.local/share/pipx/venvs
PIPX_LOG_DIR=/home/user/.local/state/pipx/log
PIPX_TRASH_DIR=/home/user/.local/share/pipx/trash
PIPX_VENV_CACHEDIR=/home/user/.cache/pipx
PIPX_STANDALONE_PYTHON_CACHEDIR=/home/user/.local/share/pipx/py
PIPX_DEFAULT_PYTHON=/home/user/local/bin/python3.12
USE_EMOJI=true
```

```sh
$ pipx environment --global
Environment variables (set by user):

PIPX_HOME=
PIPX_BIN_DIR=
PIPX_MAN_DIR=
PIPX_SHARED_LIBS=
PIPX_DEFAULT_PYTHON=
PIPX_FETCH_MISSING_PYTHON=
USE_EMOJI=

Derived values (computed by pipx):

PIPX_HOME=/opt/pipx
PIPX_BIN_DIR=/usr/local/bin
PIPX_MAN_DIR=/usr/local/share/man
PIPX_SHARED_LIBS=/opt/pipx/shared
PIPX_LOCAL_VENVS=/opt/pipx/venvs
PIPX_LOG_DIR=/opt/pipx/logs
PIPX_TRASH_DIR=/opt/pipx/.trash
PIPX_VENV_CACHEDIR=/opt/pipx/.cache
PIPX_STANDALONE_PYTHON_CACHEDIR=/opt/pipx/py
PIPX_DEFAULT_PYTHON=/usr/local/bin/python3.12
USE_EMOJI=true

```

```sh
$ pipx environment --help
usage: pipx environment [-h] [--quiet] [--verbose] [--global] [--value VARIABLE]

[...]

Available variables:
PIPX_HOME, PIPX_BIN_DIR, PIPX_MAN_DIR, PIPX_SHARED_LIBS, PIPX_LOCAL_VENVS,
PIPX_LOG_DIR, PIPX_TRASH_DIR, PIPX_VENV_CACHEDIR, PIPX_DEFAULT_PYTHON, USE_EMOJI, PIPX_HOME_ALLOW_SPACE

[...]
```

**Expected behavior**

```sh
$ pipx environment
Environment variables (set by user):

PIPX_HOME=
PIPX_GLOBAL_HOME=             # <-- added
PIPX_BIN_DIR=
PIPX_GLOBAL_BIN_DIR=          # <-- added
PIPX_MAN_DIR=
PIPX_GLOBAL_MAN_DIR=          # <-- added
PIPX_SHARED_LIBS=
PIPX_DEFAULT_PYTHON=
PIPX_FETCH_MISSING_PYTHON=
USE_EMOJI=
PIPX_HOME_ALLOW_SPACE=

Derived values (computed by pipx):

PIPX_HOME=/home/user/.local/share/pipx
PIPX_BIN_DIR=/home/user/.local/bin
PIPX_MAN_DIR=/home/user/.local/share/man
PIPX_SHARED_LIBS=/home/user/.local/share/pipx/shared
PIPX_LOCAL_VENVS=/home/user/.local/share/pipx/venvs
PIPX_LOG_DIR=/home/user/.local/state/pipx/log
PIPX_TRASH_DIR=/home/user/.local/share/pipx/trash
PIPX_VENV_CACHEDIR=/home/user/.cache/pipx
PIPX_STANDALONE_PYTHON_CACHEDIR=/home/user/.local/share/pipx/py
PIPX_DEFAULT_PYTHON=/usr/local/bin/python3.12
USE_EMOJI=true
PIPX_HOME_ALLOW_SPACE=false
```

```sh
$ pipx environment --global
Environment variables (set by user):

PIPX_HOME=
PIPX_GLOBAL_HOME=             # <-- added
PIPX_BIN_DIR=
PIPX_GLOBAL_BIN_DIR=          # <-- added
PIPX_MAN_DIR=
PIPX_GLOBAL_MAN_DIR=          # <-- added
PIPX_SHARED_LIBS=
PIPX_DEFAULT_PYTHON=
PIPX_FETCH_MISSING_PYTHON=
USE_EMOJI=
PIPX_HOME_ALLOW_SPACE=

Derived values (computed by pipx):

PIPX_HOME=/opt/pipx
PIPX_BIN_DIR=/usr/local/bin
PIPX_MAN_DIR=/usr/local/share/man
PIPX_SHARED_LIBS=/opt/pipx/shared
PIPX_LOCAL_VENVS=/opt/pipx/venvs
PIPX_LOG_DIR=/opt/pipx/logs
PIPX_TRASH_DIR=/opt/pipx/trash
PIPX_VENV_CACHEDIR=/opt/pipx/.cache
PIPX_STANDALONE_PYTHON_CACHEDIR=/opt/pipx/py
PIPX_DEFAULT_PYTHON=/usr/local/bin/python3.12
USE_EMOJI=true
PIPX_HOME_ALLOW_SPACE=false
```

```sh
$ pipx environment --help
usage: pipx environment [-h] [--quiet] [--verbose] [--global] [--value VARIABLE]

[...]

Available variables:
PIPX_HOME, PIPX_GLOBAL_HOME, PIPX_BIN_DIR, PIPX_GLOBAL_BIN_DIR,
PIPX_MAN_DIR, PIPX_GLOBAL_MAN_DIR, PIPX_SHARED_LIBS, PIPX_LOCAL_VENVS,
PIPX_LOG_DIR, PIPX_TRASH_DIR, PIPX_VENV_CACHEDIR, PIPX_DEFAULT_PYTHON,
USE_EMOJI, PIPX_HOME_ALLOW_SPACE

[...]
```

### PR DESCRIPTION
**Pull request by viscruocco**
Fixed #1492 

- [x] I have added a news fragment under `changelog.d/` (if the patch affects the end users)

## Summary of changes

* Added `PIPX_GLOBAL_HOME`, `PIPX_GLOBAL_BIN_DIR` and `PIPX_GLOBAL_MAN_DIR` to output of `pipx environment` under the `Environment variables (set by user)` section.
* Added documentation of the `PIPX_GLOBAL_*` vars in `pipx --help`, `pipx install --help`, `pipx environment --help` and `installation.md`.

## Test plan

Tested by running

```
$ pipx environment
$ pipx environment --global
$ pipx environment --help
$ pipx install --help
```

### CODE REVIEW

**Comment by chrysle** [comment]
Could you also add a test for these variables, please?

**Comment by viscruocco** [comment]
> Could you also add a test for these variables, please?

Is this minimal test expansion sufficient?
https://github.com/pypa/pipx/pull/1493/commits/2bbce3cf0187c28687f8f12354773675a7635be8

**Comment by viscruocco** [comment]
> I'd like to also test the paths, though not strictly necessary.

Actually there are no paths to check here, because the variables are just added to the `set by user` section, which is not populated in the tests at all:

$ pipx environment --global
Environment variables (set by user):

PIPX_HOME=
PIPX_GLOBAL_HOME=             # <-- added
PIPX_BIN_DIR=
PIPX_GLOBAL_BIN_DIR=          # <-- added
PIPX_MAN_DIR=
PIPX_GLOBAL_MAN_DIR=          # <-- added
PIPX_SHARED_LIBS=
PIPX_DEFAULT_PYTHON=
PIPX_FETCH_MISSING_PYTHON=
USE_EMOJI=
PIPX_HOME_ALLOW_SPACE=

Derived values (computed by pipx):

PIPX_HOME=/opt/pipx
PIPX_BIN_DIR=/usr/local/bin
PIPX_MAN_DIR=/usr/local/share/man
PIPX_SHARED_LIBS=/opt/pipx/shared
PIPX_LOCAL_VENVS=/opt/pipx/venvs
PIPX_LOG_DIR=/opt/pipx/logs
PIPX_TRASH_DIR=/opt/pipx/trash
PIPX_VENV_CACHEDIR=/opt/pipx/.cache
PIPX_STANDALONE_PYTHON_CACHEDIR=/opt/pipx/py
PIPX_DEFAULT_PYTHON=/usr/local/bin/python3.12
USE_EMOJI=true
PIPX_HOME_ALLOW_SPACE=false

**Review by Gitznik — APPROVED** [review]
This looks good to me. Also nice improvement on the docs themselves :+1:

**Comment by chrysle** [comment]
> Actually there are no paths to check here

Yep, I thought about creating a separate test and populating these variables manually, to verify they take effect.

**Comment by viscruocco** [comment]
> Yep, I thought about creating a separate test and populating these variables manually, to verify they take effect.

I'm sorry, I think this is a slightly bigger change than what I'm in for right now. Hope that's okay.

File: src/pipx/main.py [hunk 1]
```diff
@@ -908,8 +920,10 @@ def _add_environment(subparsers: argparse._SubParsersAction, shared_parser: argp
             variables and platform specific default values.
 
             Available variables:
-            PIPX_HOME, PIPX_BIN_DIR, PIPX_MAN_DIR, PIPX_SHARED_LIBS, PIPX_LOCAL_VENVS,
-            PIPX_LOG_DIR, PIPX_TRASH_DIR, PIPX_VENV_CACHEDIR, PIPX_DEFAULT_PYTHON, USE_EMOJI, PIPX_HOME_ALLOW_SPACE
+            PIPX_HOME, PIPX_GLOBAL_HOME, PIPX_BIN_DIR, PIPX_GLOBAL_BIN_DIR,
+            PIPX_MAN_DIR, PIPX_GLOBAL_MAN_DIR, PIPX_SHARED_LIBS, PIPX_LOCAL_VENVS,
+            PIPX_LOG_DIR, PIPX_TRASH_DIR, PIPX_VENV_CACHEDIR, PIPX_DEFAULT_PYTHON,
+            USE_EMOJI, PIPX_HOME_ALLOW_SPACE
```
Comment (chrysle) [inline src/pipx/main.py:926]: Keeping this list up to date seems tedious. Couldn't we import `pipx.commands.environment.environment_variables` and join via f-string, as done in other places?

**Comment by chrysle** [comment]
> I'm sorry, I think this is a slightly bigger change than what I'm in for right now. I hope it's okay if I leave it at that.

Sure, no problem!

File: src/pipx/main.py (hunk 1 above)
Comment (viscruocco) [reply src/pipx/main.py]: Good idea! Done in [3ab8d48](https://github.com/pypa/pipx/pull/1493/commits/3ab8d4893ae1e05ff465fb220e7c076db4078ec7)
I also used this approach to automatically test the presence of all documented environment variables in the output of `pipx environment`.

**Review by chrysle — APPROVED** [review]

**Comment by chrysle** [comment]
LGTM, thanks! Please rebase or merge.

**Pull request merged**